composio/sdk/sdk.py

```diff
@@ -312,84 +313,75 @@ def get_app(self, app_name: str):
                 "We just pass the important actions to the agent, but this is not meant "
                 "to be used in production. Check out https://docs.composio.dev/sdk/python/actions for more information.",
                 UserWarning,
             )
-            actions = actions_response["items"]
-            important_tag = Tag.IMPORTANT.value[1]
-            important_actions = [
-                action
-                for action in actions
-                if important_tag in (action.get("tags", []) or [])
-            ]
-            if len(important_actions) > 5:
-                return important_actions
-            else:
-                return actions
+
+        all_relevent_app_ids = app_unique_ids + [action.value[0] for action in actions]
+        all_relevent_app_ids = list(set(all_relevent_app_ids))
+
+        action_response = self.http_handler.get_action_schemas(
+            app_unique_ids=all_relevent_app_ids, use_case=use_case, limit=limit
+        )
+
+        action_schema_list = action_response["items"]
+        filtered_actions = []
+        important_actions = []
+        important_tag = Tag.IMPORTANT.value[1]
+
+        if tags is None:
+            tags = []
+        elif isinstance(tags, Tag):
+            tags = [tags]
+            
+        if Tag.ALL in tags:  # or all_relevent_app_ids == []:
+            # Everything should be added when Tag.ALL is specifically mentioned, or All actions actions and apps are called.
+            tag_values = []
+        elif not tags:
+            # Only add the important actions If no tag is given or if number of important actions are more more than 5
+            tag_values = [Tag.IMPORTANT.value[1]]
         else:
-            raise ValueError("Either apps or actions must be provided")
+            if actions:
+                raise ValueError("Both actions and tags cannot be provided together")
+            tag_values = [tag.value[1] if isinstance(tag, Tag) else tag for tag in tags]
+
+        for action_schema in action_schema_list:
+            if action_schema["appName"] in app_unique_ids or all_relevent_app_ids == []:
+                if tag_values == [] or tag_values == [Tag.IMPORTANT.value[1]]:
+                    filtered_actions.append(action_schema)
+                elif bool(set(tag_values) & set(action_schema["tags"])):
+                    filtered_actions.append(action_schema)
+
+                if important_tag in action_schema["tags"]:
+                    important_actions.append(action_schema)
+
+            elif action_schema["name"] in action_unique_ids:
+                filtered_actions.append(action_schema)
+                important_actions.append(action_schema)
+
+        if (tags == [Tag.IMPORTANT]) or (
+            len(important_actions) > 5
+            and tag_values == [Tag.IMPORTANT.value[1]]
+            and all_relevent_app_ids != []
+        ):
+            return important_actions
```

Comment:
Basically if I give tag = Tag.REPOS, then this wouldn't be invoked right?

Reply:
No, This is only for the case when Tag.Important is specifically mentioned, or no tag is mentioned with atleast one app. @kaavee315